drivers/spi/spi_context.h

```diff
@@ -19,23 +19,32 @@
 extern "C" {
 #endif
 
+enum spi_ctx_runtime_op_mode {
+	SPI_CTX_RUNTIME_OP_MODE_MASTER = BIT(0),
+	SPI_CTX_RUNTIME_OP_MODE_SLAVE  = BIT(1),
+};
+
 struct spi_context {
-	struct spi_config *config;
+	const struct spi_config *config;
```

Comment:
The standard way of using the context is to copy here the pointer passed to `transceive`. But one may decide to call `transceive` with a pointer to a structure that is an automatic variable, i.e. located on the stack. In case of a synchronous operation it is irrelevant, but for an asynchronous one it can be really destructive. This should be at least documented somewhere, in descriptions of asynchronous parts of SPI API probably.

Reply:
Indeed it requires to be documented. spi_config should not be located on the stack.